client/dashboard/components/text/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { __experimentalText as WPText } from '@wordpress/components';
+import clsx from 'clsx';
+import { forwardRef } from 'react';
+import './style.scss';
+
+export interface TextProps extends React.ComponentProps< typeof WPText > {
+	intent?: 'success' | 'warning' | 'error';
+}
+
+function UnforwardedText(
+	{ intent, ...props }: TextProps,
+	ref: React.ForwardedRef< HTMLElement >
+) {
+	return (
+		<WPText
+			ref={ ref }
+			{ ...props }
+			className={ clsx(
+				{
+					'dashboard-text--success': intent === 'success',
+					'dashboard-text--warning': intent === 'warning',
+					'dashboard-text--error': intent === 'error',
+				},
```

Comment:
Why not

```suggestion
				intent && `dashboard-text--${ intent }`
```

Seems like we've already type checked that it is a valid string value.